src/core/algorithms/cind/condition_miners/cinderella.cpp

```diff
@@ -0,0 +1,156 @@
+#include "cinderella.h"
+
+#include <algorithm>
+#include <iterator>
+#include <list>
+#include <map>
+#include <tuple>
+#include <unordered_map>
+
+#include "cind/condition.h"
+#include "cind/condition_miners/basket.h"
+#include "cind/condition_type.h"
+
+namespace algos::cind {
+namespace {
+std::list<BasketInfo> MergeBaskets(std::list<BasketInfo> const& baskets1,
+                                   std::list<BasketInfo> const& baskets2) {
+    auto it1 = baskets1.begin();
+    auto it2 = baskets2.begin();
+    std::list<BasketInfo> result;
+    while (it1 != baskets1.end() && it2 != baskets2.end()) {
+        auto const& [index1, positions1, is_included1] = *it1;
+        auto const& [index2, positions2, is_included2] = *it2;
+        if (index1 < index2) {
+            ++it1;
+        } else if (index1 > index2) {
+            ++it2;
+        } else {
+            std::list<size_t> positions_intersection;
+            std::set_intersection(positions1.begin(), positions1.end(), positions2.begin(),
+                                  positions2.end(), std::back_inserter(positions_intersection));
+            if (!positions_intersection.empty()) {
+                result.emplace_back(index1, positions_intersection, is_included1);
```

Comment:
Here you can use std::move()
```suggestion
                result.emplace_back(index1, std::move(positions_intersection), is_included1);
```